src/integrators.jl

```diff
@@ -1,152 +1,275 @@
+import DataStructures
+
 """
     DistributedODEIntegrator <: AbstractODEIntegrator
 
 A simplified variant of [`ODEIntegrator`](https://github.com/SciML/OrdinaryDiffEq.jl/blob/6ec5a55bda26efae596bf99bea1a1d729636f412/src/integrators/type.jl#L77-L123).
 
 """
-mutable struct DistributedODEIntegrator{algType,uType,tType,pType} <: DiffEqBase.AbstractODEIntegrator{algType,true,uType,tType}
-    prob
+mutable struct DistributedODEIntegrator{algType, uType, tType, pType} <:
+               DiffEqBase.AbstractODEIntegrator{algType, true, uType, tType}
     alg::algType
     u::uType
     p::pType
-    dt::tType
     t::tType
-    tstop::tType
-    tdir::tType
+    dt::tType
+    _dt::tType # copy of argument to __init used to set dt in step!
+    dtchangeable::Bool
+    tstops
+    _tstops # copy of argument to __init used as a default argument to reinit!
+    saveat
+    _saveat # copy of argument to __init used as a default argument to reinit!
```

Comment:
Shouldn't these be typed?

Reply:
Maybe this is why ClimaTimeSteppers is failing inference?

Reply:
Ah, yes, every integrator field is typed in OrdinaryDiffEq. I'll go ahead and add type parameters for everything.